cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -340,38 +339,37 @@ func (s *Supervisor) getBootstrapInfo() (err error) {
 
 func (s *Supervisor) Capabilities() protobufs.AgentCapabilities {
 	var supportedCapabilities protobufs.AgentCapabilities
-	if c := s.config.Capabilities; c != nil {
-		// ReportsEffectiveConfig is set if unspecified or explicitly set to true.
-		if (c.ReportsEffectiveConfig != nil && *c.ReportsEffectiveConfig) || c.ReportsEffectiveConfig == nil {
-			supportedCapabilities |= protobufs.AgentCapabilities_AgentCapabilities_ReportsEffectiveConfig
-		}
 
-		// ReportsHealth is set if unspecified or explicitly set to true.
-		if (c.ReportsHealth != nil && *c.ReportsHealth) || c.ReportsHealth == nil {
-			supportedCapabilities |= protobufs.AgentCapabilities_AgentCapabilities_ReportsHealth
-		}
+	c := s.config.Capabilities
 
-		// ReportsOwnMetrics is set if unspecified or explicitly set to true.
-		if (c.ReportsOwnMetrics != nil && *c.ReportsOwnMetrics) || c.ReportsOwnMetrics == nil {
-			supportedCapabilities |= protobufs.AgentCapabilities_AgentCapabilities_ReportsOwnMetrics
-		}
+	if c.ReportsEffectiveConfig {
```

Comment:
What do you think about putting this section inside a `(Capabilities).GetSupportedCapabilities` method? That way adding a new capability can be done in a single place.

Reply:
Moved this to capabilities and added tests; it seemed like we weren't setting the ReportsStatus bit, which according to the docs is required, so I added that to the logic while I was writing the tests.